x-pack/test/api_integration/deployment_agnostic/apis/observability/ai_assistant/complete/functions/retrieve_elastic_doc.spec.ts

```diff
@@ -112,25 +125,65 @@ export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderCon
         });
       });
 
-      it('makes 6 requests to the LLM', () => {
-        expect(llmProxy.interceptedRequests.length).to.be(6);
+      it('makes 2 requests to the LLM', () => {
+        expect(llmProxy.interceptedRequests.length).to.be(2);
       });
 
-      it('every request contain retrieve_elastic_doc function', () => {
-        const everyRequestHasRetrieveElasticDoc = llmProxy.interceptedRequests.every(
-          ({ requestBody }) =>
-            requestBody.tools?.some((t) => t.function.name === 'retrieve_elastic_doc')
-        );
-        expect(everyRequestHasRetrieveElasticDoc).to.be(true);
+      it('emits 5 messageAdded events', () => {
+        expect(messageAddedEvents.length).to.be(5);
       });
 
-      it('contains the original user message', () => {
-        const everyRequestHasUserMessage = llmProxy.interceptedRequests.every(({ requestBody }) =>
-          requestBody.messages.some(
-            (message) => message.role === 'user' && (message.content as string) === USER_MESSAGE
-          )
-        );
-        expect(everyRequestHasUserMessage).to.be(true);
+      describe('The first request', () => {
+        it('contains the retrieve_elastic_doc function', () => {
+          expect(firstRequestBody.tools?.map((t) => t.function.name)).to.contain(
+            'retrieve_elastic_doc'
+          );
+        });
+
+        it('leaves the LLM to choose the correct tool by leave tool_choice as auto and passes tools', () => {
+          expect(firstRequestBody.tool_choice).to.be('auto');
+          expect(firstRequestBody.tools?.length).to.not.be(0);
+        });
+      });
+
+      describe('The second request - Sending the user prompt', () => {
+        let lastMessage: ChatCompletionMessageParam;
+        let parsedContent: { documents: Array<{ title: string; content: string; url: string }> };
+        before(() => {
+          lastMessage = last(secondRequestBody.messages) as ChatCompletionMessageParam;
+          parsedContent = JSON.parse(lastMessage.content as string);
+        });
+        it('includes the retrieve_elastic_doc function call', () => {
+          expect(secondRequestBody.messages[4].role).to.be(MessageRole.Assistant);
+          // @ts-expect-error
+          expect(secondRequestBody.messages[4].tool_calls[0].function.name).to.be(
+            'retrieve_elastic_doc'
+          );
+        });
+
+        it('responds with the correct tool message', () => {
+          expect(lastMessage?.role).to.be('tool');
+          // @ts-expect-error
+          expect(lastMessage?.tool_call_id).to.equal(
+            // @ts-expect-error
+            secondRequestBody.messages[4].tool_calls[0].id
+          );
```

Comment:
`lastMessage` is defined above: `lastMessage = last(secondRequestBody.messages)`

So this is testing:

```ts
          expect(last(secondRequestBody.messages)?.tool_call_id).to.equal(
            // @ts-expect-error
            secondRequestBody.messages[4].tool_calls[0].id
          );
```

Is that the intention?